### DISCUSSION TITLE
[$50 bounty] How to get list of backups for a volume programmatically? API?

### DISCUSSION THREAD

**Discussion by tobiasmuehl** [longhorn/longhorn#4640] [Q&A] [answered]
I'd like to feed the backup status of longhorn volumes into an external alerting system. This could be achieved through a JSON API, though I don't think one exists for Longhorn? Maybe it can also be achieved using the kubernetes API?

The goal is to monitor the backup system of Longhorn to get an alert if a backup fails for any reason. I have an existing alerting system that is already closely monitored and would prefer to use that vs a built-in alerting system in longhorn, since the monitoring system itself also needs to be monitored.

**Comment by tobiasmuehl** [comment, marked as answer]
I'll send the person that can answer this $50 USD.

Edit: This was answered in comments, see below

**Reply by c3y1huang** [reply]
Does Longhorn [backup](https://longhorn.io/docs/1.3.1/monitoring/metrics/#backup) metrics work for you?

Ref https://longhorn.io/docs/1.3.1/monitoring/prometheus-and-grafana-setup/

**Reply by tobiasmuehl** [reply]
It can be a part of the solution, but from what I can tell it can only be used to get metrics about *a single* backup, right? I need something to find me every backup of every volume

**Reply by tobiasmuehl** [reply]
It could work. I can get the actual size and the state of every backup in the system. It doesn't allow me to find out how old a backup is, but if I have an external system that queries prometheus regularly, I can at least estimate when the backup was created. Does it sound possible? @c3y1huang

**Reply by c3y1huang** [reply]
If the query is regular enough, yes I think you can have a good estimate of when the backup was created.

And BTW If your alerting/monitoring system is able to handle websocket, Longhorn has [this](https://github.com/longhorn/longhorn-manager/blob/master/api/router.go#L209-L211) available for the UI to stream the backups, ex: ws://1.2.3.4:<sve-longhorn-frontend>/v1/ws/1s/backups.

**Reply by tobiasmuehl** [reply]
Wow, the websocket URL I would've never found. That looks like what I needed. Do you want to receive the bounty via your BuyMeACoffee link in your Github profile? @c3y1huang

**Reply by tobiasmuehl** [reply]
Sent. https://www.buymeacoffee.com/c3y1huang/c/4061491

<img width="1242" alt="Screenshot 2022-10-03 at 21 58 20" src="https://user-images.githubusercontent.com/3341826/193609713-562b26f2-4543-4f9e-b35a-3bea5d28ec17.png">
